### PR TITLE
Reduce usage of SubProgressMonitor

### PR DESCRIPTION
**Pull request by vogella**

### CODE REVIEW

File: resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/Policy.java [hunk 1]
```diff
@@ -179,10 +184,6 @@ public static IProgressMonitor monitorFor(IProgressMonitor monitor) {
 	}
 
 	public static IProgressMonitor subMonitorFor(IProgressMonitor monitor, int ticks) {
-		if (monitor == null)
-			return new NullProgressMonitor();
-		if (monitor instanceof NullProgressMonitor)
-			return monitor;
-		return new SubProgressMonitor(monitor, ticks);
+		return SubMonitor.convert(monitor, ticks);
```
Comment (HeikoKlare) [inline resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/Policy.java:187]: Is this utility method really necessary after the simplification? Couldn't that code just be inlined?

Comment (vogella) [reply resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/Policy.java:187]: Yes, SubMonitor.convert can also handle NullProgressMonitor IIRC so the whole thing can be further reduced.

Comment (vogella) [reply resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/Policy.java:187]: > Yes, SubMonitor.convert can also handle NullProgressMonitor IIRC so the whole thing can be further reduced.

Ups, I did that already...

Comment (vogella) [reply resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/Policy.java:187]: Looking at the "inline" I would prefer not to do this via refactoring, as it looks to me that the code logic can be further simplified, e.g. removing done calls, simplifying the call. I do a few simplifications with another PR, maybe others want to join the effort to simplify the code here.

**Comment by vogella** [comment]
Looks like it is not that simple, as the caller call multiple times the beginTask method....

**Pull request closed without merging**